### PR TITLE
Artemis 447

### PR DESCRIPTION
**Pull request by grs**

### CODE REVIEW

**Comment by mtaylor** [comment]
@grs I'm not sure what happened here, but it seems there's some commits from your local branch or something.  Could you squash or resend with just the single commit please.

**Comment by clebertsuconic** [comment]
@mtaylor @grs The script we use to merge will fix it.

File: artemis-protocols/artemis-proton-plug/src/main/java/org/proton/plug/context/AbstractConnectionContext.java [hunk 1]
```diff
@@ -73,6 +76,8 @@ public AbstractConnectionContext(AMQPConnectionCallback connectionCallback,
                                     ScheduledExecutorService scheduledPool) {
       this.connectionCallback = connectionCallback;
       this.containerId = (containerId != null) ? containerId : UUID.randomUUID().toString();
+      connectionProperties.put(Symbol.valueOf("product"), "apache-activemq-artemis");
```
Comment (johnament) [inline artemis-protocols/artemis-proton-plug/src/main/java/org/proton/plug/context/AbstractConnectionContext.java]: Should this be overrideable in some way?

File: tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/proton/ProtonTest.java [hunk 2]
```diff
@@ -211,6 +213,25 @@ public void testBrokerContainerId() throws Exception {
    }
 
    @Test
+   public void testBrokerConnectionProperties() throws Exception {
+      if (protocol != 0 && protocol != 3) return; // Only run this test for AMQP protocol
```
Comment (johnament) [inline tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/proton/ProtonTest.java]: Can this be an assume statement (this way the test will be marked as skipped if this is not true)

Comment (mtaylor) [inline tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/proton/ProtonTest.java]: Yes.  It'd be better to use an AssumeTrue or similar.   We do this in many other areas of the test class, so perhaps we can update the to use the junit assertions in a separate commit.

File: artemis-protocols/artemis-proton-plug/src/main/java/org/proton/plug/context/AbstractConnectionContext.java (hunk 1 above)
Comment (grs) [inline artemis-protocols/artemis-proton-plug/src/main/java/org/proton/plug/context/AbstractConnectionContext.java]: The version string is overrideable I believe so if you had some custom patched version you could indicate that. The brokers in Qpid don't make this value customisable, so I didn't personally think it was needed. I have no objection though if others have a use case for altering that.

**Comment by grs** [comment]
@mtaylor sorry, I think that was because I merged rather than rebased when syncing my fork. Will avoid that next time. 

@clebertsuconic thanks!

**Pull request closed without merging**